### ISSUE TITLE
Add support for ignoring files

### ISSUE THREAD

**Issue by rwjblue** [ember-template-lint/ember-template-lint#1803]
Sometimes you want to specifically ignore files from being linted (for example, you have some server side handlebars files in the repo).

I'd like for these ignores to go into the existing `.template-lintrc.js` file (I don't _really_ want to have "yet another file" for this like `.eslintignore`). 

Perhaps something like:

```js
module.exports = {
  extends: 'recommended',
  
  ignoredFiles: [.....],
};
```

Thoughts?

**Comment by bmish** [comment]
If we were to add this option, we should probably follow the lead of .eslintrc.js and call it `ignoredPatterns`: https://eslint.org/docs/user-guide/configuring/ignoring-code#ignorepatterns-in-config-files

However, I would note that having a CLI option and an ignore file (like `.templatelintignore`) are both commonly supported for the purpose of ignoring files by other linters:
* https://eslint.org/docs/user-guide/configuring/ignoring-code#the-eslintignore-file
* https://npmpackagejsonlint.org/docs/en/cli#--ignorepath-alias--i
* https://stylelint.io/user-guide/ignore-code#files-entirely

**Comment by rwjblue** [comment]
@bmish - To confirm, you are suggesting that we do all three of:

* add `ingoredPatterns` to existing config file
* add `.template-lintignore` (😭 at the dash, but what the heck was I thinking at the beginning)
* add `--ignore-path` CLI option

Right?

(to be clear, I'd be happy with that result)

**Comment by bmish** [comment]
Yes. ESLint has a large variety of ways for ignoring files, and while I don't think it's necessarily needed or urgent to support all of them, my preference would be to follow ESLint's lead.

**Comment by rwjblue** [comment]
Yep, I'm sold.

**Comment by lolmaus** [comment]
I've added a TypeDoc theme to my codebase. TypeDoc happens to use Handlebars. Now `ember-template-lint` freaks out. :(
